clayui.com/src/utils/getSection.js

```diff
@@ -61,42 +57,32 @@ const toSectionElements = (
 	};
 };
 
-const toSectionItem = (item, paths) => {
-	const isParentPackage = Object.keys(navigationRelationships).includes(
-		`/${item.pathname}`
-	);
-
-	if (item.navigationParent) {
-		navigationRelationships[item.navigationParent].push(
-			`/${item.pathname}`
-		);
-	}
-
+const toSectionItem = (item, paths, parentPaths) => {
 	if (item.isFolder) {
 		item.items = paths
-			.filter((path) => path.link !== item.link)
+			.filter(path => path.link !== item.link)
 			.filter(
-				(path) =>
+				path =>
 					path.link ===
 					item.parentLink + path.id + (path.isFolder ? '/index' : '')
 			)
-			.filter((path) => !path.navigationParent)
-			.map((path) => toSectionItem(path, paths))
+			.filter(path => !path.navigationParent)
+			.map(path => toSectionItem(path, paths, parentPaths))
 			.sort(sortByOrderAndTitle);
 	}
 
-	if (isParentPackage) {
+	if (parentPaths.includes(item.pathname)) {
 		item.items = paths
-			.filter((path) => path.navigationParent)
-			.filter((path) => path.link !== item.link)
-			.map((path) => toSectionItem(path, paths))
+			.filter(path => path.navigationParent === item.pathname)
```

Comment:
This check ensures that only the items with that navigationParent get added to the right path, otherwise they get added to every parent.